### ISSUE TITLE
Inquiry about data construction

### ISSUE THREAD

**Issue by Ray0202** [zhefu2/SerenEnhance#1]
For example

I find some code like

```
# prepare data path
user_data_path = '../data/user_pre_encoded_sorted' # item embeddings
rel_path = '../data/user_rel_train'  # relevance samples
unp_path = '../data/user_unexpectedness_samples'  # unexpectedness samples
sre_path = '../data/user_pre_candidate' # serendipity samples
test_path = '../data/user_pre_test'  # test set
```

However, there is no way to figure out how to construct this data, so I hope the owner can complete this code or provide clear instructions in the documentation.

**Comment by zhefu2** [comment]
Thanks for your question. The training data was not uploaded to GitHub due to its size. Instead, it has been provided in a shared Google Drive folder, as detailed in the README file (in the Quick Start). You can access it directly using this link: https://drive.google.com/drive/folders/10zNYgfyoe5NVJM3uPUZTTuOEDzG7sHuu?usp=sharing. 

The details of data construction are presented in our paper. Please refer to Section 4.2 for item embeddings, Section 4.3 for relevance and unexpectedness samples, Section 5.1 for serendipity samples, and Section 5.3 for the test set.

If there are any other questions, please don't hesitate to let me know.

**Comment by stellafull** [comment]
Hi, Zhe Fu, I noticed 
`unp_path = '../data/user_unexpectedness_samples'  # unexpectedness samples` 
is not provided in Google Drive Link, and also there's some bugs in the Unexpectedness_generation.py and unable to reproduce the data/user_unexpectedness_samples, can U update py code or upload data, it would be much helpful.

**Comment by zhefu2** [comment]
Let me review the code over the next few days, and will update it soon. I did not upload the "user_unexpectedness_samples" since the data already exceeded my Google Drive quota.
